cmd/validators/main.go

```diff
@@ -120,12 +127,159 @@ type pubKeyInfo struct {
+			keyFl    = flag.String("key", "", "Hex encoded, private key that transaction should be signed with.")
+		)
+		flag.Parse()
+
+		if *keyFl == "" {
+			return errors.New("private key is required")
+		}
+		key, err := decodePrivateKey(*keyFl)
+		if err != nil {
+			return fmt.Errorf("cannot decode private key: %s", err)
+		}
+
+		raw, err := ioutil.ReadAll(in)
+		if err != nil {
+			return fmt.Errorf("cannot read transaction from stdin: %s", err)
+		}
+		if len(raw) == 0 {
+			return errors.New("no stdin data")
+		}
+		var tx app.Tx
+		if err := tx.Unmarshal(raw); err != nil {
+			return fmt.Errorf("cannot deserialize transaction: %s", err)
+		}
+
+		genesis, err := fetchGenesis(*tmAddrFl)
+		if err != nil {
+			return fmt.Errorf("cannot fetch genesis: %s", err)
+		}
+
+		bnsClient := client.NewClient(client.NewHTTPConnection(*tmAddrFl))
+		aNonce := client.NewNonce(bnsClient, key.PublicKey().Address())
+		if seq, err := aNonce.Next(); err != nil {
+			return fmt.Errorf("cannot get the next sequence number: %s", err)
+		} else {
+			client.SignTx(&tx, key, genesis.ChainID, seq)
+		}
+
+		if raw, err := tx.Marshal(); err != nil {
+			return fmt.Errorf("cannot serialize transaction: %s", err)
+		} else {
+			out.Write(raw)
+		}
+		return nil
+	}
+}
+
+func cmdMultisigSubmit(in io.Reader) func() error {
+	return func() error {
+		flag.Usage = func() {
+			fmt.Fprint(flag.CommandLine.Output(), `
+Read binary serialized validator change transaction from standard input and
+submit it.
+
+This command is intended to be used to submit multisig authenticated requests.
+Make sure to collect enough signatures before submitting.
+
+`)
+			flag.PrintDefaults()
+		}
+		var (
+			tmAddrFl = flag.String("tm", "https://bns.NETWORK.iov.one:443", "Tendermint node address. Use proper NETWORK name.")
+		)
+		flag.Parse()
+
+		raw, err := ioutil.ReadAll(in)
+		if err != nil {
+			return fmt.Errorf("cannot read transaction from stdin: %s", err)
+		}
+		if len(raw) == 0 {
+			return errors.New("no stdin data")
+		}
+		var tx app.Tx
+		if err := tx.Unmarshal(raw); err != nil {
+			return fmt.Errorf("cannot deserialize transaction: %s", err)
+		}
+		bnsClient := client.NewClient(client.NewHTTPConnection(*tmAddrFl))
+
+		if err := bnsClient.BroadcastTx(&tx).IsError(); err != nil {
+			return fmt.Errorf("cannot broadcast transaction: %s", err)
+		}
+		return nil
+	}
+}
+
 func cmdAdd() error {
 	var (
 		tmAddrFl = flag.String("tm", "https://bns.NETWORK.iov.one:443", "Tendermint node address. Use proper NETWORK name.")
 		pubKeyFl = flag.String("pubkey", "", "Base64 encoded, ed25519 public key.")
 		hexKeyFl = flag.String("key", "", "Hex encoded, private key of the validator that is to be added/updated.")
-		powerFl  = flag.Int64("power", 10, "Validator node power. Set to 0 to delete a node.")
+		powerFl  = flag.Int64("power", 2, "Validator node power. Set to 0 to delete a node.")
```

Comment:
q: why 2? currently 10 is default in the testnets? The use should enter the value rather than us setting a default value, IMHO.

Reply:
:man_shrugging: Changing back to 10.